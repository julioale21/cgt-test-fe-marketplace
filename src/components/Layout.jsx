import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
