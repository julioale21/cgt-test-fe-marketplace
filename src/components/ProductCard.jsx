import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
