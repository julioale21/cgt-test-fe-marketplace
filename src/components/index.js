export { ContainedButton } from "./buttons/ContainedButton";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { ProductList } from "./ProductList";
export {ProductCard} from "./ProductCard";
