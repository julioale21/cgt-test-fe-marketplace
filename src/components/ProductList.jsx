import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
