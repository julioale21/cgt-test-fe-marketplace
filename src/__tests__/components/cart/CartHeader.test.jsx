import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { CartHeader } from '../../../components';

describe('CartHeader', () => {
  const defaultProps = {
    itemCount: 1,
    onClearCart: jest.fn()
  };

  it('should match snapshot when displaying a single item', () => {
    const props = { ...defaultProps };

    const { container } = render(<CartHeader {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when displaying multiple items', () => {
    const props = { ...defaultProps, itemCount: 3 };

    const { container } = render(<CartHeader {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should display item count in singular form', () => {
    const props = { ...defaultProps };

    render(<CartHeader {...props} />);

    expect(screen.getByText('Shopping Cart (1 items)')).toBeInTheDocument();
  });

  it('should display item count in plural form', () => {
    const props = { ...defaultProps, itemCount: 3 };

    render(<CartHeader {...props} />);

    expect(screen.getByText('Shopping Cart (3 items)')).toBeInTheDocument();
  });

  it('should call onClearCart when clear button is clicked', async () => {
    const props = { ...defaultProps };
    const user = userEvent.setup();

    render(<CartHeader {...props} />);
    await user.click(screen.getByText('Clear Cart'));

    expect(props.onClearCart).toHaveBeenCalledTimes(1);
  });

  it('should render clear cart button with small size', () => {
    const props = { ...defaultProps };

    render(<CartHeader {...props} />);
    const button = screen.getByRole('button', { name: 'Clear Cart' });

    expect(button).toHaveClass('MuiButton-sizeSmall');
  });
});
