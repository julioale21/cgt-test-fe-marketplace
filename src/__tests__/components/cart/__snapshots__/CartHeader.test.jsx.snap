// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader should match snapshot when displaying a single item 1`] = `
<div>
  <div
    class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
      >
        <div
          class="MuiBox-root css-69i1ev"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Shopping Cart (
            1
             items)
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-133o6ou-MuiButtonBase-root-MuiButton-root"
            id=":r0:"
            tabindex="0"
            type="button"
          >
            Clear Cart
          </button>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`CartHeader should match snapshot when displaying multiple items 1`] = `
<div>
  <div
    class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
      >
        <div
          class="MuiBox-root css-69i1ev"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
          >
            Shopping Cart (
            3
             items)
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-133o6ou-MuiButtonBase-root-MuiButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            Clear Cart
          </button>
        </div>
      </span>
    </div>
  </div>
</div>
`;
